relay-server/src/actors/project.rs

```diff
@@ -781,8 +781,12 @@ impl Project {
             return;
         }
 
-        self.state_channel = None;
-        self.state = Some(state.clone());
+        match self.expiry_state() {
+            // If the new state is invalid but the old one still usable, keep the old one.
+            ExpiryState::Updated(old) | ExpiryState::Stale(old) if state.invalid() => state = old,
+            // If the new state is valid or the old one is expired, always use the new one.
+            _ => self.state = Some(state.clone()),
```

Comment:
Is there a chance both states are updated? What should we do there?

Reply:
It's a good question. We _could_ write this function in a way that it always preserves the fresher config, but for now I would keep the default behavior of replacing the old state with the state passed as an argument, and only make an exception for invalid configs.